common/ast/navigable_test.go

```diff
@@ -120,6 +135,44 @@ func TestNavigateAST(t *testing.T) {
 	}
 }
 
+func TestNavigableExprChildren(t *testing.T) {
```

Comment:
Could we also test for post order traversal?

Reply:
The post order ends up getting tested indirectly via `MaxID`, but I can add an explicit test to show it within this one.

Reply:
The method is called in MaxID but I didn't come across any tests which validate the actual visiting order. Regardless, I think an explicit test is valuable.

Reply:
Agreed. I chose to show pre and post explicitly with an example pre order using `Children` as well.